src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -264,12 +339,20 @@ public async Task<TResult> SendRequestAsync<TResult>(JsonRpcRequest request, Can
             throw new McpClientException("Transport is not connected");
         }
 
+        using Activity? activity = s_activitySource.StartActivity("SendingRequest");
+
         // Set request ID
         if (request.Id.IsDefault)
         {
             request.Id = new RequestId($"{_id}-{Interlocked.Increment(ref _nextRequestId)}");
         }
 
+        if (activity is not null)
```

Comment:
```suggestion
        if (activity is { IsAllDataRequested: true })
```

The tags are effective only when `AllDataRequested` so maybe we could avoid adding tags if not necessary

see https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.isalldatarequested